include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h

```diff
@@ -681,6 +681,18 @@ class ThreadPoolTempl : public onnxruntime::concurrency::ExtendedThreadPoolInter
 
   ThreadPoolProfiler profiler_;
 
+  void SignalAllAndWait() {
+    done_ = true;
+
+    // Now if all threads block without work, they will start exiting.
+    // But note that threads can continue to work arbitrary long,
+    // block, submit new work, unblock and otherwise live full life.
+    WakeAllWorkersForExit();
```

Comment:
>WakeAllWorkersForExit(); [](http://example.com/codeflow?start=4&length=24)

We have a problem here, that not all of the worker_data entries has a real thread

Reply:
Not an issue